test/integration/crud/explain.test.ts

```diff
@@ -117,6 +119,111 @@ describe('CRUD API explain option', function () {
       });
     }
   }
+
+  describe('explain helpers w/ maxTimeMS', function () {
+    let client: MongoClient;
+    const commands: CommandStartedEvent[] = [];
+    let collection: Collection;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({}, { monitorCommands: true });
+      await client.connect();
+
+      await client.db('explain-test').dropDatabase();
+      collection = await client.db('explain-test').createCollection('bar');
+
+      client.on('commandStarted', filterForCommands('explain', commands));
+      commands.length = 0;
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    describe('cursor explain commands', function () {
+      describe('when maxTimeMS is specified via a cursor explain method, it sets the property on the command', function () {
+        test('find()', async function () {
+          await collection
+            .find({ name: 'john doe' })
+            .explain({ maxTimeMS: 2000, verbosity: 'queryPlanner' });
+
+          const [{ command }] = commands;
+          expect(command).to.have.property('maxTimeMS', 2000);
+        });
+
+        test('aggregate()', async function () {
+          await collection
+            .aggregate([{ $match: { name: 'john doe' } }])
+            .explain({ maxTimeMS: 2000, verbosity: 'queryPlanner' });
+
+          const [{ command }] = commands;
+          expect(command).to.have.property('maxTimeMS', 2000);
+        });
+      });
+
+      it('when maxTimeMS is not specified, it is not attached to the explain command', async function () {
+        await collection.find({ name: 'john doe' }).explain({ verbosity: 'queryPlanner' });
+
+        const [{ command }] = commands;
+        expect(command).not.to.have.property('maxTimeMS');
+      });
+
+      it('when maxTimeMS is specified as an explain option and a command-level option, the explain option takes precedence', async function () {
+        await collection
+          .find(
+            {},
+            {
+              maxTimeMS: 1000,
+              explain: {
+                verbosity: 'queryPlanner',
+                maxTimeMS: 2000
+              }
+            }
+          )
+          .toArray();
+
+        const [{ command }] = commands;
+        expect(command).to.have.property('maxTimeMS', 2000);
+      });
```

Comment:
There should probably be tests for what happens when `maxTimeMS` is only specified as a command-level option, and not as an explain option at all